pennylane/templates/state_preparations/basis_qutrit.py

```diff
@@ -112,7 +114,18 @@ def compute_decomposition(basis_state, wires):  # pylint: disable=arguments-diff
         """
 
         op_list = []
+
+        if qml.math.is_abstract(basis_state):
+            tshift = qml.math.array([[0, 0, 1], [1, 0, 0], [0, 1, 0]])
+            for wire, state in zip(wires, basis_state):
+                mat = qml.math.linalg.matrix_power(tshift, state)
+                op = qml.ops.QutritUnitary(mat, wires=wire)
```

Comment:
So the workflow for computing gradients has a different decomposition which is not hardware-friendly?